.github/workflows/docker-images.yaml

```diff
@@ -66,17 +66,28 @@ jobs:
           ${{ inputs.semver-latest != 'none' && format('--semver-latest {0}', inputs.semver-latest) || '' }}
 
       - name: Test with Temporalite
-        run: docker run --rm -i -v /tmp/temporal-certs:/certs sdk-features:${{ inputs.lang }}-${{ inputs.run-tag }}
+        run: docker run --rm -i -v /tmp/temporal-certs:/certs ${{ env.SDK_FEAT_BUILT_IMAGE_TAG }}
 
       - name: Test with Cloud
         # Only supported in non-fork runs
         if: ${{ github.event.pull_request.head.repo.full_name == '' || github.event.pull_request.head.repo.full_name == 'temporalio/sdk-features' }}
         run: |
-          docker run --rm -i -v /tmp/temporal-certs:/certs sdk-features:${{ inputs.lang }}-${{ inputs.run-tag }} \
+          docker run --rm -i -v /tmp/temporal-certs:/certs ${{ env.SDK_FEAT_BUILT_IMAGE_TAG }} \
           --server $TEMPORAL_CLOUD_ADDRESS \
           --namespace $TEMPORAL_CLOUD_NAMESPACE \
           --client-cert-path /certs/client.pem \
           --client-key-path /certs/client.key
 
-      # TODO(bergundy): create a docker push step (only run under certain conditions - like manual trigger or main
-      # branch trigger)
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        if: inputs.do-push
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Push image to DockerHub
+        if: inputs.do-push
+        # TODO: Push all tags that we built
+        run: |
+          docker tag ${{ env.SDK_FEAT_BUILT_IMAGE_TAG }} temporaliotest/sdk-features:${{ inputs.lang }}
```

Comment:
I need to get this merged to be able to test it unfortunately. You can't run manually triggered GH workflows that aren't in the default branch, although, amusingly, once they are you can point at a different branch's implementation to use. So I'll finish this step in another PR.

Reply:
Good to know.
I guess it's enough to create a dummy workflow on the main branch and test it against the development branch.